cypress/test/search-field.spec.ts

```diff
@@ -0,0 +1,94 @@
+import SearchFieldSteps from "../../cypress/steps/search-field-steps";
+
+describe('Onto Search Field Component', () => {
+  it('Should find the component', () => {
+    SearchFieldSteps.visit();
+    SearchFieldSteps.getSearchFieldComponent().should('be.visible');
+  });
+
+  it('Should autocomplete', () => {
+    SearchFieldSteps.visit();
+    SearchFieldSteps.typeSearchPhrase('D');
+    SearchFieldSteps.getAutocompleteLabel(0).should('be.visible').and('contain', 'Dehydroepiandrosterone');
+    SearchFieldSteps.getAutocompleteType(0).should('be.visible').and('contain', 'small molecule');
+    SearchFieldSteps.getAutocompleteLabels(0).should('be.visible').and('contain', 'Prasterone,5-Dehydroepiandrosterone,DHEA,5-DHEA,3-beta-Hydroxy-5-androsten-17-one,DHA');
+  });
+
+  it('Should make visual difference when free text or autocomplete is selected', () => {
+    SearchFieldSteps.visit();
+    SearchFieldSteps.typeSearchPhrase('D');
+    SearchFieldSteps.getAutocompleteLabel(0).click();
+    SearchFieldSteps.completeSearchPhrase('free text');
+
+    SearchFieldSteps.getStateSelection(0).should('be.visible')
+      .should('contain', 'Dehydroepiandrosterone')
+      .and('have.class', 'state-suggestion');
+
+    SearchFieldSteps.getStateSelection(1).should('be.visible')
+      .should('contain', 'free text')
+      .and('have.class', 'state-free-text');
+  });
+
+  it('Should delete selected', () => {
+    SearchFieldSteps.visit();
+    SearchFieldSteps.typeSearchPhrase('D');
+    SearchFieldSteps.getAutocompleteLabel(0).click();
+    SearchFieldSteps.completeSearchPhrase('free text');
+    SearchFieldSteps.deleteStateSelection(0);
+
+    SearchFieldSteps.getStateSelection(0).should('be.visible')
+      .should('contain', 'free text')
+
+    SearchFieldSteps.getStateSelection(1).should('not.be.visible');
+  });
+
+  it('Should not duplicate selected', () => {
+    SearchFieldSteps.visit();
+    SearchFieldSteps.typeSearchPhrase('D');
+    SearchFieldSteps.getAutocompleteLabel(0).should('contain', 'Dehydroepiandrosterone').click();
+
+    SearchFieldSteps.getStateSelection(0).should('be.visible').and('contain', 'Dehydroepiandrosterone');
+
+    SearchFieldSteps.typeSearchPhrase('D');
+    SearchFieldSteps.getAutocompleteLabel(0).should('contain', 'Dehydroepiandrosterone').click();
+    SearchFieldSteps.getStateSelection(0).should('be.visible').and('contain', 'Dehydroepiandrosterone');
+
+    SearchFieldSteps.getStateSelection(1).should('not.be.visible');
```

Comment:
uhmm this is kind of confuing. Why don't you check for selections size instead?

Reply:
done